### PR TITLE
chore: Refactor AdyenActionComponent to use one actionComponent at a time

### PR DESCRIPTION
**Pull request by kaphacius**

### CODE REVIEW

File: Adyen.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4754,6 +4757,7 @@
 				F917613825A30B5700D653BE /* ThreeDS2ComponentFingerprint.swift in Sources */,
 				F970143925D139BE007E74D9 /* VoucherComponentStyle.swift in Sources */,
 				F917613925A30B5700D653BE /* ThreeDS2ComponentFingerprintToken.swift in Sources */,
+				5A2D1951267368580082BCE9 /* AdyenActionComponentStyle.swift in Sources */,
```
Comment (mohammedDehairy) [inline Adyen.xcodeproj/project.pbxproj:4760]: can this be just `ActionComponentStyle`, adyen is sort of redundant.

File: AdyenActions/AdyenActionComponent.swift [hunk 2]
```diff
@@ -114,10 +103,10 @@ public final class AdyenActionComponent: ActionComponent, Localizable {
     }
     
     private func handle(_ action: ThreeDS2ChallengeAction) {
-        guard let threeDS2Component = threeDS2Component else { return }
+        guard let threeDS2Component = currentActionComponent as? ThreeDS2Component else { return }
```
Comment (mohammedDehairy) [inline AdyenActions/AdyenActionComponent.swift:106]: and assertionFailure here? "ThreeDS2Component is nil, There must be a `ThreeDS2FingerprintAction` action preceding a `ThreeDS2ChallengeAction` action"

File: AdyenActions/AdyenActionComponent.swift [hunk 3]
```diff
@@ -25,24 +25,20 @@ public final class AdyenActionComponent: ActionComponent, Localizable {
...
-    /// Indicates the UI configuration of the await component.
-    public var awaitComponentStyle: AwaitComponentStyle?
-
-    /// Indicates the UI configuration of the voucher component.
-    public var voucherComponentStyle: VoucherComponentStyle?
-
-    /// Indicates the UI configuration of the QR code component.
-    public var qrCodeComponentStyle: QRCodeComponentStyle?
+    /// :nodoc:
+    public let style: AdyenActionComponentStyle
```
Comment (mohammedDehairy) [inline AdyenActions/AdyenActionComponent.swift:29]: can document this in the MIGRATION.md file?

File: Adyen.xcodeproj/project.pbxproj (hunk 1 above)
Comment (kaphacius) [reply Adyen.xcodeproj/project.pbxproj:4760]: 👍

File: AdyenActions/AdyenActionComponent.swift (hunk 2 above)
Comment (kaphacius) [reply AdyenActions/AdyenActionComponent.swift:106]: missed this during cherry-picking, thanks!

**Review by descorp — APPROVED** [review]

**Review by mohammedDehairy — APPROVED** [review]

File: AdyenDropIn/Components/PaymentMethodListComponent.swift [hunk 4]
```diff
@@ -58,7 +58,6 @@ internal final class PaymentMethodListComponent: ComponentLoader, PresentableCom
             listItem.imageURL = LogoURLProvider.logoURL(for: component.paymentMethod, environment: apiContext.environment)
             listItem.trailingText = displayInformation.disclosureText
             listItem.subtitle = displayInformation.subtitle
-            listItem.showsDisclosureIndicator = showsDisclosureIndicator
```
Comment (descorp) [inline AdyenDropIn/Components/PaymentMethodListComponent.swift:61]: local variable `showsDisclosureIndicator ` still remains

**Review by descorp — APPROVED** [review]

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**